src/main/java/com/cryptomorin/xseries/XMaterial.java

```diff
@@ -2151,12 +2120,29 @@ private boolean isDuplicated() {
          *
          * @since 1.0.0
          */
-        private static final int VERSION = Integer.parseInt(getMajorVersion(Bukkit.getVersion()).substring(2));
+        private static final int VERSION = parseVersion(Bukkit.getVersion());
         /**
          * Cached result if the server version is after the v1.13 flattening update.
          *
          * @since 3.0.0
          */
         private static final boolean ISFLAT = supports(13);
+
+        /**
+         * Gets the exact major version (..., 9, 10, ..., 14)
+         *
+         * @param version Supports the version as provided by {@link Bukkit#getVersion()}
+         *
+         * @return the exact major version.
+         * @see #VERSION
+         * @since 8.5.0
+         */
+        private static int parseVersion(String version) {
+            Matcher matcher = Pattern.compile("MC: \\d\\.(\\d+)").matcher(version);
```

Comment:
move this pattern.compile(...) code to constant field.

Reply:
@portlek CryptoMoring asked me to avoid caching the compiled regex, so I decided I should go this way. What way is preferred?

Reply:
java allocates a lot of resources to compile patterns, many java developer is caching pattern unless it's dynamic. in this case, `MC: \\d\\.(\\d+)` is just a static pattern so move it to class like
`private static final Pattern VERSION_PATTERN = Pattern.compile("MC: \\d\\.(\\d+)");`

Reply:
Thanks, resolved.

Reply:
I have no idea what @portlek is talking about. Versions are not parsed multiple times, you're just wasting memory and gaining no performance whatsoever doing this.

Reply:
I've reverted the commit.